lib/routing/sfRoute.class.php

```diff
@@ -230,16 +230,7 @@ public function generate($params, $context = [], $absolute = false)
             throw new InvalidArgumentException($message);
         }
 
-        $elements = array_merge(
-            [SharedCacheHelper::ROUTING_NAMESPACE],
-            $tparams
```

Comment:
This is the source of the problem - only the _values_ in `$tparams` were being included in the cache key generation.

The result of this oversight meant that we'd get incorrect cache hits for multiple different instances of `url_for(...)`.  For example:
- `url_for('pack', ['id' => $order_id, 'confirmation' => 1, 'force' => 1])`
- `url_for('pack', ['id' => $order_id, 'undigital-flyer' => 1, 'force' => 1])`

Both of these calls to `url_for()` would result in different final routes being generated if we were not caching. However, according to the previous caching logic, the cache keys for both these routes (and any in a similar format) would look something like: `pack/[order_id]/1/1`.

Whichever route had been cached under that key first would be what got returned for all similarly-formatted routes for the next three months, which is obviously nonsense.